app/packages/core/src/components/Grid/useSelect.ts

```diff
@@ -24,11 +30,62 @@ export default function useSelect(
         }
 
         retained.add(id.description);
+
+        const newFieldsIfAny = getNewFields(id.description);
+
+        const overlays =
+          instance instanceof VideoLooker
+            ? instance.pluckedOverlays
+            : instance.sampleOverlays;
```

Comment:
I don't think exposing these is necessary. `refreshSample` could capture the overlay filtering logic happening below